packages/fuels-abigen-macro/src/lib.rs

```diff
@@ -30,6 +34,102 @@ pub fn wasm_abigen(input: TokenStream) -> TokenStream {
         .into()
 }
 
+/// This proc macro is used to reduce the amount of boilerplate code in integration tests.
+/// When expanded, the proc macro will: launch a local provider, generate one wallet,
+/// deploy the selected contract and create a contract instance.
+/// The names for the contract instance and wallet variables must be provided as inputs.
+/// This macro can be called multiple times inside a function if the variables names are changed.
+/// The same contract can be deployed multiple times as the macro uses deployment with salt.
+/// If however; you need to have a shared wallet between macros, the first macro must set the
+/// wallet name to `shared_wallet`. The other ones must set the wallet name to `None`.
+#[proc_macro]
+pub fn setup_contract_test(input: TokenStream) -> TokenStream {
+    let args = parse_macro_input!(input as Spanned<ContractTestArgs>);
+
+    let abs_forc_dir = Path::new(&args.project_path)
+        .canonicalize()
+        .unwrap_or_else(|_| {
+            panic!(
+                "Unable to canonicalize forc project path: {}. Make sure the path is valid!",
+                &args.project_path
+            )
+        });
+
+    let forc_project_name = abs_forc_dir.file_name().unwrap().to_str().unwrap();
```

Comment:
Nit to avoid `unwrap` in order to help us debugger down the line 😄 

```suggestion
    let forc_project_name = abs_forc_dir.file_name().expect("failed to get filename").to_str().expect("failed to convert filename to string");
```